lib/types.ml

```diff
@@ -25,14 +25,19 @@ module Opam = struct
     let pp = pp_sexp sexp_of_t
   end
 
-  type repo =
-    [ `Github of string * string
-    | `Git of string
-    | `Unknown of string
-    | `Virtual
-    | `Error of string ]
+  type repo = [ `Github of string * string | `Git of string | `Virtual | `Error of string ]
   [@@deriving sexp]
 
+  let equal_repo repo repo' =
+    match (repo, repo') with
+    | `Github (user, repo), `Github (user', repo') ->
+        String.equal user user' && String.equal repo repo'
+    | `Git s, `Git s' | `Error s, `Error s' -> String.equal s s'
+    | `Virtual, `Virtual -> true
+    | (`Github _ | `Git _ | `Unknown _ | `Virtual | `Error _), _ -> false
```

Comment:
You removed the `Unknown` tag.

Reply:
Good catch. This showcases why we should use regular variants here, especially since we don't take advantage of their polymorphic nature anywhere in the code.